py/test/selenium/webdriver/firefox/ff_profile_tests.py

```diff
@@ -171,3 +171,52 @@ def test_autodetect_proxy_is_set_in_profile():
 
     profile.set_proxy(proxy)
     assert profile.default_preferences["network.proxy.type"] == ProxyType.AUTODETECT['ff_value']
+
+
+def test_add_extension_web_extension_with_id(capabilities, webserver):
```

Comment:
These tests are just for legacy Firefox, and we should also add coverage to `py/test/selenium/webdriver/marionette`

Reply:
It seems that many tests for legacy Firefox are missing for marionette. Should this be reported in general as a separate issue?

Reply:
I'd like to see more unit tests for the Selenium client that don't require a driver, for example we should be able to test several aspects of the `FirefoxProfile` class that aren't affected by the version of Firefox. Where there is a need for a driver test, we should at least make sure we have coverage in modern Firefox (marionette) as we will eventually be looking to drop support for legacy Firefox. If you're interested in contributing more, I'd love to get a coverage report for our unit tests and work on improving it - this will also give us confidence when we come to do larger scale refactoring.